ext/nnstreamer/tensor_filter/tensor_filter_support_cc.cc

```diff
@@ -0,0 +1,53 @@
+/**
+ * GStreamer Tensor_filter, C++ Subplugin Support. (this is not a subplugin)
+ * Copyright (C) 2020 MyungJoo Ham <myungjoo.ham@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_filter_support_cc.cc
+ * @date	22 Jan 2020
+ * @brief	Base class for tensor_filter subplugins of C++ classes.
+ * @see		http://github.com/nnsuite/nnstreamer
+ * @author	MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * @details
+ *    This is not a subplugin, but a helper for C++ subplugins.
+ *    If you want to write a wrapper for neural network frameworks
+ *    or a hardware adpator in C++, this is what you want.
+ *    If you want to attach a single C++ class/object as a filter
+ *    (a.k.a. custom filter) to a pipeline, use tensor_filter_cpp
+ */
+#include "tensor_filter_support_cc.hh"
+
+/** @todo The code is incomplete. NYI! */
+bool
+tensor_filter_subplugin::isDimFuncOverriden (void)
+{
+  typedef int (*getDimPtr)(GstTensorInfo *);
+  typedef int (*setDimPtr)(const GstTensorsInfo *, GstTensorInfo *);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpmf-conversions"
+  bool isOverridenGetInput (getDimPtr)((*this).*(&tensor_filter_subplugin::getInputDimension)) !=
+      (getDimPtr)(&tensor_filter_subplugin::getInputDimension);
+  bool isOverridenGetOutput (getDimPtr)((*this).*(&tensor_filter_subplugin::getOutputDimension)) !=
+      (getDimPtr)(&tensor_filter_subplugin::getOutputDimension);
+  bool isOverridenSetInput (setDimPtr)((*this).*(&tensor_filter_subplugin::setInputDimension)) !=
+      (setDimPtr)(&tensor_filter_subplugin::setInputDimension);
+#pragma GCC diagnostic pop
+  if (isOverridenGetInput && isOverridenGetOutput)
+    return true;
+  if (isOverridenSetOutput)
```

Comment:
`isOverridenSetInput`?

Reply:
With your C-class update, this will be rewritten, but.. yes.. that should be updated.